docs/tutorials/mbed_os_2_vs_5.md

```diff
@@ -0,0 +1,65 @@
+## Arm Mbed OS 2 vs. Arm Mbed OS 5
+
+Arm Mbed OS 5 is a superset of Arm Mbed OS 2. This means that Mbed OS 5 *is* Mbed OS 2 but with more features, such as an RTOS.
+
+### Overview
+
+- Mbed OS 5 is a *superset* of Mbed OS 2.
+- Mbed OS 5 is a combination of both Mbed OS 2 and Mbed-RTOS.
+- Mbed OS 2 and Mbed OS 5 both use the same `.h` file.
```

Comment:
Another big difference worth mentioning: Drivers are thread-safe in Mbed OS 5, and are not thread-safe in Mbed OS 2. This implies that: Mbed OS 5 is more than Mbed OS 2 and Mbed-RTOS